userdata.tmpl

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+add-apt-repository -y ppa:wireguard/wireguard
+apt-get update -y
+apt-get upgrade -y
+apt-get install nginx wireguard -y
+wget https://raw.githubusercontent.com/nginxinc/NGINX-Demos/master/nginx-hello/index.html --output-document /usr/share/nginx/html/index.html
+wget https://raw.githubusercontent.com/nginxinc/NGINX-Demos/master/nginx-hello/hello.conf --output-document /etc/nginx/sites-enabled/default
+systemctl restart nginx
+mkdir /etc/wireguard/keys
+cd /etc/wireguard/keys
+umask 077
+wg genkey | tee privatekey | wg pubkey > publickey
```

Comment:
I tested it out and looks good. I was unable to add ppa but it wasn't necessary as wireguard is already maintained in the distro http://us-east-1.ec2.archive.ubuntu.com/ubuntu

```
Cannot add PPA: 'ppa:~wireguard/ubuntu/wireguard'.
The team named '~wireguard' has no PPA named 'ubuntu/wireguard'
Please choose from the following available PPAs:
```